db/nft.go

```diff
@@ -426,24 +443,28 @@ func GetCreators(conn *pgxpool.Conn, q QueryCreatorRequest, p PageRequest) (res
 			'count', count)),
 		COUNT(*) OVER() AS row_count
 	FROM (
-		SELECT i.owner, i.iscn_id_prefix, c.class_id, SUM(%s.latest_price) AS value, COUNT(DISTINCT n.id) as count
+		SELECT i.owner, i.iscn_id_prefix, c.class_id, SUM(%[1]s) AS value, COUNT(DISTINCT n.id) as count
 		FROM iscn AS i
 		JOIN iscn_latest_version
 		ON i.iscn_id_prefix = iscn_latest_version.iscn_id_prefix
 			AND ($5 = true OR i.version = iscn_latest_version.latest_version)
 		JOIN nft_class AS c ON i.iscn_id_prefix = c.parent_iscn_id_prefix
 		JOIN nft AS n ON c.class_id = n.class_id
 			AND ($4::text[] IS NULL OR cardinality($4::text[]) = 0 OR n.owner != ALL($4))
+		JOIN nft_event AS e 
+		ON e.nft_id = n.nft_id
+			AND e.receiver = n.owner
```

Comment:
In most cases this would only result in 1 event, but it is possible that more than 1 event can be returned if the nft is then sold and bought again by same account. Would this cause issue?

Reply:
Yes, this would be an issue, handled with a subquery filter.

Reply:
Does the subquery always select the whole nft event table though, performance issue?

Reply:
Should creating an index for `id, nft_id, receiver` help?

Reply:
it would be still requiring whole table without any where condition every time we call this API
But on second though, is multiple purchase event counted towards a sum really an issue in this case?
The user would really have paid twice actually

Reply:
If we don't filter with the `latest_e`, one drawback is the total value will be inflated. 
For example, if `A` brought an NFT with a current price of `1024 LIKE` in `32 LIKE`, and send it back and forth to his other account `B` twice. 
His `total value priced by class` would be `1024 x 3 = 3072 LIKE` but not `1024 LIKE`. 
The same issue happens on `total value priced by NFT` if he brought it back rather than just sending it back from another account.
Not sure if this is a concern.

Reply:
Hmm keep this subquery and let's see it's performance on prod data then

Reply:
Should also use lateral subquery here

Reply:
Got missed, was in the joy of newfound, thanks for reminding!